tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/management/ActiveMQServerControlTest.java

```diff
@@ -4290,6 +4296,106 @@ public boolean isStarted() {
       }
    }
 
+   @Test
+   public void testManualStopStartEmbeddedWebServer() throws Exception {
+      FakeWebServerComponent fake = new FakeWebServerComponent();
+      server.addExternalComponent(fake, true);
+      Assert.assertTrue(fake.isStarted());
+
+      ActiveMQServerControl serverControl = createManagementControl();
+      serverControl.stopEmbeddedWebServer();
+      Assert.assertFalse(fake.isStarted());
+      serverControl.startEmbeddedWebServer();
+      Assert.assertTrue(fake.isStarted());
+   }
+
+   @Test
+   public void testRestartEmbeddedWebServer() throws Exception {
+      FakeWebServerComponent fake = new FakeWebServerComponent();
+      server.addExternalComponent(fake, true);
+      Assert.assertTrue(fake.isStarted());
+
+      ActiveMQServerControl serverControl = createManagementControl();
+      long time = System.currentTimeMillis();
+      Assert.assertTrue(time >= fake.getStartTime());
+      Assert.assertTrue(time > fake.getStopTime());
+      Thread.sleep(5);
+      serverControl.restartEmbeddedWebServer();
+      Assert.assertTrue(serverControl.isEmbeddedWebServerStarted());
+      Assert.assertTrue(time < fake.getStartTime());
+      Assert.assertTrue(time < fake.getStopTime());
+   }
+
+   @Test
+   public void testRestartEmbeddedWebServerTimeout() throws Exception {
+      final CountDownLatch startDelay = new CountDownLatch(1);
+      FakeWebServerComponent fake = new FakeWebServerComponent(startDelay);
+      server.addExternalComponent(fake, false);
+
+      ActiveMQServerControl serverControl = createManagementControl();
+      try {
+         serverControl.restartEmbeddedWebServer(10);
+         fail();
+      } catch (Exception e) {
+         Assert.assertTrue(e instanceof ActiveMQIllegalStateException);
+      }
+      startDelay.countDown();
```

Comment:
The countdown should go in a finally or it might never happen. Rather than catching Exception it could just catch the expected exception type and let everything else bubble up (then you also dont need the assertion).

Reply:
Done.